### PR TITLE
feat: openPaymentPage that calls initiate,process together using Hype…

### PR DESCRIPTION
**Pull request by namitgoel**
openPaymentPage that calls initiate,process together using HyperCheckoutLite openPaymentPage

### CODE REVIEW

File: www/hypersdk.js [hunk 1]
```diff
@@ -40,5 +40,9 @@ module.exports = {
     },
     isInitialised: function (callback) {
         callNative("isInitialised", {}, callback);
-    }
+    },
+    openPaymentPage: function (payload, callback) {
+        sharedCallback = callback;
```
Comment (harshgarg18) [inline www/hypersdk.js:45]: why are we assigning callback to sharedCallback here? This is not required.

File: src/android/HyperSDKPlugin.java [hunk 2]
```diff
@@ -191,6 +203,45 @@ public void preFetch(JSONArray args) {
...
+                        LogSubCategory.LifeCycle.HYPER_SDK,
+                        LogLevel.ERROR,
+                        SDK_TRACKER_LABEL,
+                        "initiate",
+                        "activity is null"
+                    );
+                    return;
+                }
+                JSONObject params = new JSONObject(String.valueOf(args.get(0)));
+                HyperCheckoutLite.openPaymentPage(activity, params, new HyperPaymentsCallbackAdapter() {
```
Comment (harshgarg18) [inline src/android/HyperSDKPlugin.java:222]: We need to do this in a separate activity, as is done for the process call in the plugin.

CC: @Subhash703

**Review by harshgarg18 — CHANGES_REQUESTED** [review]

**Comment by harshgarg18** [comment]
- [ ] @namitgoel - Can we get this tested before merge?

File: src/android/HyperSDKPlugin.java [hunk 3]
```diff
@@ -280,19 +320,38 @@ private void process(JSONArray args) {
...
-                    LogSubCategory.LifeCycle.HYPER_SDK,
-                    LogLevel.ERROR,
-                    SDK_TRACKER_LABEL,
-                    "process",
-                    "hyperServices is null");
-            return;
-        }
         HyperSDKPlugin.processCallback = processCallback;
+        if(isHyperCheckoutLiteInteg){
+            HyperCheckoutLite.openPaymentPage(activity, params, new HyperPaymentsCallbackAdapter() {
```
Comment (Subhash703) [inline src/android/HyperSDKPlugin.java:325]: Why are we calling openPaymentPage in processWithActivity?

File: src/android/build-extras.gradle [hunk 4]
```diff
@@ -19,7 +19,7 @@ static def mostRecentVersion(List<String> versions) {
 }
 
 static def getHyperSDKVersion(rootProject) {
-    def hyperSdkVersion = '2.1.13'
+    def hyperSdkVersion = '2.1.20-rc.06'
```
Comment (harshgarg18) [inline src/android/build-extras.gradle:22]: Update SDK version here to 2.1.20-rc.10, or preferably, 2.1.20 once it is released by today/tmrw

File: src/android/HyperSDKPlugin.java [hunk 5]
```diff
@@ -191,6 +203,34 @@ public void preFetch(JSONArray args) {
         }
     }
 
+    private void openPaymentPage(JSONArray args) {
+        synchronized (lock) {
+            try {
+                isHyperCheckoutLiteInteg = true;
```
Comment (harshgarg18) [inline src/android/HyperSDKPlugin.java:209]: We need to reset this flag after process_result, right?

File: src/android/HyperSDKPlugin.java [hunk 6]
```diff
@@ -191,6 +203,34 @@ public void preFetch(JSONArray args) {
...
+                        LogLevel.ERROR,
+                        SDK_TRACKER_LABEL,
+                        "initiate",
+                        "activity is null"
+                    );
+                    return;
+                }
+                String params = String.valueOf(args.get(0));
+                JSONObject payloadParams = new JSONObject(params);
+                if (isPPMerchant(payloadParams)) {
```
Comment (harshgarg18) [inline src/android/HyperSDKPlugin.java:223]: is this condition really need? The function is called `openPaymentPage`!

**Review by harshgarg18 — CHANGES_REQUESTED** [review]
The code is not properly formatted as well.

**Pull request merged**